[UserPage] Show Brew's Meta thumbnail on UserPage

G-Ambatte opened the pull request:
This PR resolves #2331.

This PR adds the thumbnail metadata to the information passed to the UserPage, and adds new elements to the BrewItem when that information is present. 

Currently the result looks like this:

![image](https://user-images.githubusercontent.com/19826659/188270177-d8bd1ce7-db39-41d1-8325-b56cb7e82baf.png)

calculuschild commented on client/homebrew/pages/basePages/listPage/brewItem/brewItem.jsx:
```diff
@@ -98,6 +98,12 @@ const BrewItem = createClass({
+				<div className='thumbnail'>
+					<img className='thumbnailImage' src={brew.thumbnail} ></img>
+				</div> :
+				<></>
```
Do we need this empty fragment?

calculuschild replied:
I think you can do something like 

```
{brew.thumbnail && <div ....
```

So it draws the div if true, and just doesn't do anything if false.

calculuschild commented on client/homebrew/pages/basePages/listPage/brewItem/brewItem.less:
```diff
@@ -19,6 +19,20 @@
+		float: left;
+		margin-right: 5px;
+		display: flex;
+		img.thumbnailImage {
+			max-width: 100%;
+			max-height: 100%;
+			margin: auto;
```
Similar behavior but grows the image to fill the whole box and crops the excess instead of shrinking to fit. Might look a little cleaner instead of the empty space inside the box.

```suggestion
		img.thumbnailImage {
			width : 100%;
			height : 100%;
			object-fit : cover;
			border-radius : inherit;
```

calculuschild commented on client/homebrew/pages/basePages/listPage/brewItem/brewItem.less:
```diff
@@ -19,6 +19,21 @@
 	box-shadow                  : 0px 4px 5px 0px #333;
 	background-color            : #cab2802e;
+	.thumbnail  {
+		width: 80px;
+		height: 100px;
```
What do you think about making this a square? 80px x 80px?  My guess the average image is closer to a square, but I actually have no idea.

calculuschild commented on client/homebrew/pages/basePages/listPage/brewItem/brewItem.less:
```diff
@@ -19,6 +19,21 @@
+		float: left;
+		margin-right: 5px;
+		display: flex;
+		img.thumbnailImage {
```
There is some slight positioning weirdness where the image is not quite filling the whole box. I'm not sure why, but there must be some tweak to make that fit perfectly. 
![image](https://user-images.githubusercontent.com/5878534/188298231-4d6db1c2-e3bd-47e8-a7c6-1d4688fe8166.png)

ericscheid replied:
This can be fixed by removing the `border` and `border-radius` from the container div, and applying instead directly on the image (assuming `object-fit: cover;` is also being used)

Gazook89 commented:
In my opinion, image should be on the right side of the brew item. Otherwise the titles are not aligned between items with images and not

Gazook89 commented:
Here is a quick mockup of an alternate idea....again, no local install so just did this in devtools for now:

<img width="337" alt="image" src="https://user-images.githubusercontent.com/58999374/189709363-fb404363-8d1b-4013-b347-71922ae7622b.png">

```css
element {
	background-image: url(https://i.imgur.com/YPV3Txc.jpg);
	position: absolute;
	width: 150px;
	height: 100%;
	top: 0;
	right: 0;
	z-index: -1;
	background-size: cover;
	mask-image: linear-gradient(80deg, #0000 20%, #050 40%);
	opacity: 50%;
}
```

Will it work for 100% of images?  probably not...the efficacy will depend largely on the composition of the image used.

calculuschild commented:
Not a bad idea. How does it look with a light image?

Gazook89 commented:
> @Gazook89 not a bad idea. How does it look with a light image?

Here is a pretty light image, also demonstrates the use of an image with a white background:
<img width="336" alt="image" src="https://user-images.githubusercontent.com/58999374/189712558-e482f238-2c37-449d-bbd2-dd59e675d7f3.png">

I think it looks *good enough*.  One issue is that the fade removes the head of the creature.  This is the danger of any clipping, though-- one must take account of the image's composition.  In my mind this is an *annoyance* but one that is easily overlooked for now.  Some day I think the entire user page could be reworked to something besides a two column scroll-a-thon and when that happens the images could be incorporated better-- but that is a different Issue.

Perhaps a default image of the HB logo could be used, as well, if no thumbnail is present...or maybe that is unwanted clutter.

---
Here is the html node in case anyone wants to quickly copy/paste into devtools again:
```html
<div style="background-image: url(https://www.aidedd.org/dnd/images/bone-devil.jpg);position: absolute;width: 150px;height: 100%;top: 0;right: 0;z-index: -1;background-size: cover;mask-image: linear-gradient(80deg, #0000 20%, #050 40%);opacity: 50%;"></div>
```

calculuschild commented:
@G-Ambatte What do you think? Do you want to try applying @Gazook89 's style and move the image to the right?

G-Ambatte commented:
> @G-Ambatte What do you think? Do you want to try applying @Gazook89 's style and move the image to the right?

I'm more than happy to implement the design; I'll try to push through an update to the PR later tonight.

G-Ambatte commented:
Here's a screengrab of the updated style:

![image](https://user-images.githubusercontent.com/19826659/191432314-d03fe266-6b0f-4b44-956f-7bebaa77372b.png)

G-Ambatte commented:
Some examples with no description text:

![image](https://user-images.githubusercontent.com/19826659/191441259-431c322b-a00c-41f8-8186-045ad7562c3d.png)

calculuschild commented:
This looks good to me. My only remaining thought is perhaps we should set a max-height on the image so it is guaranteed to fit into the smallest `brewItem` height.

Gazook89 commented:
> max-height

If the max height is set to the smallest brewItem height, say 200px or whatever it is, then if the brewItem height is larger it is likely that there will be an unflattering straight edge at the bottom of the image, since it won't expand to fill the brewItem.

`background-size` could be set to `contain` instead?

G-Ambatte commented:
Switched to `contain`, added `no-repeat` and positioned to `top right`. Those same four brews:

![image](https://user-images.githubusercontent.com/19826659/191597357-f54ef255-a52b-4895-922f-a27921c73a9f.png)

![image](https://user-images.githubusercontent.com/19826659/191597400-090fff12-d45a-4f33-8e0c-4080da4f5b6f.png)

calculuschild commented:
Noice. 🎉 💯 🥇
